retrymanager_getallreplicas.go

```diff
@@ -0,0 +1,56 @@
+package gocbcorex
+
+import (
+	"errors"
+	"time"
+
+	"github.com/couchbase/gocbcorex/memdx"
+)
+
+type RetryManagerGetAllReplicas struct {
+	calc BackoffCalculator
+}
+
+func NewRetryManagerGetAllReplicas() *RetryManagerGetAllReplicas {
+	return &RetryManagerGetAllReplicas{
+		calc: ExponentialBackoff(10*time.Millisecond, 500*time.Millisecond, 2),
+	}
+}
+
+func (m *RetryManagerGetAllReplicas) NewRetryController() RetryController {
+	return &retryControllerGetAllReplicas{
+		parent: m,
+	}
+}
+
+type retryControllerGetAllReplicas struct {
```

Comment:
Why do we need a separate retry controller for GetAllReplicas?

Reply:
I did this because for the individual replica reads we want to retry for `ErrNoServerAssigned` as this is what we get during a rebalance is ongoing, and the normal retry manager fails on this error.

Reply:
In fairness, we should probably retry this error generally I think.

Reply:
I think that by the definition of how GetReplicas works, I don't think it ever makes sense to return duplicates from the same node, so it probably makes sense to collapse that behaviour into the replica code itself, rather than being part of the retry manager.  Additionally, the `ErrRepeatedReplicaRead` error only appears during GetReplicas anyways, so having it in the default retry path wouldn't have any impacts.  There is already precedence for having operation-specific errors be exposed as unique errors for the purposes of having a generalized retry system as well.